crates/starknet/src/services/add_declare_transaction.rs

```diff
@@ -292,6 +301,24 @@ mod tests {
         }
     }
 
+    #[test]
+    fn add_declare_v1_transaction_second_time_should_fail() {
+        let (mut starknet, sender) = setup(None);
+
+        let declare_txn = test_declare_transaction_v1(sender);
+        starknet.add_declare_transaction_v1(declare_txn.clone()).unwrap();
+
+        let err = starknet.add_declare_transaction_v1(declare_txn).err().unwrap();
+
+        if let starknet_types::error::Error::TransactionError(
+            TransactionError::ClassAlreadyDeclared(_),
+        ) = err
+        {
+        } else {
+            panic!("Wrong error type");
+        }
```

Comment:
I spent too much time deciphering this, wouldn't the plain old `match` be clearer? 
```suggestion
        match err {
            starknet_types::error::Error::TransactionError(
                TransactionError::ClassAlreadyDeclared(_),
            ) => (),
            _ => panic!("Wrong error type"),
        }
```

Reply:
Removed the test in [b1dfbba](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/39/commits/b1dfbba19216a4e961fa2719ca41f68a833ec883)